controllers/cronset_controller.go

```diff
@@ -152,6 +153,15 @@ func (r *CronSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	currentDependentCronJobCount, err := r.getDependentCronJobCount(ctx, cronSet.Name)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.updateStatus(cronSet, currentDependentCronJobCount); err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
This would prevent the status from updating when the previous error occurs.
What do you think about using `defer`? I think I can update other statuses like `NumberMisscheduled` without difficulty.

Reply:
That's such a good idea! but that's why I suggested we need to change the logic such as immediately throwing error during creating cronjob.